### PR TITLE
[RFC] implement highlighting for output:highlight

### PR DESCRIPTION
**Pull request by blankname**
It's currently a bit slow.
What would be the best way for me to cache the output of the highlight command (instead of calling it three times)?

If this sort of approach to highlighting is acceptable, I plan to implement highlighting for at least output:let and output:call.

### CODE REVIEW

File: rplugin/python3/denite/source/output.py [hunk 1]
```diff
@@ -17,6 +17,55 @@ def __init__(self, vim):
...
+            ' deniteSource_outputHighlight_' + hi + 'xxx' +
+            ' ' + hi
+        )
+        self.vim.command(hi_name)
+        self.vim.command(hi_xxx)
+
+    def define_syntax(self):
+        cmd = self.context['args'][0]
+        if re.fullmatch(r'hi(ghlight)?(!)?', cmd):
+            self.vim.command('syntax include syntax/vim.vim')
```
Comment (Shougo) [inline rplugin/python3/denite/source/output.py]: I think it should be in `self.deine_syntax_for_hi()`.
`define_syntax()` should not be long.

**Comment by Shougo** [comment]
> What would be the best way for me to cache the output of the highlight command (instead of calling it three times)?

Nothing.  Unfortunately.
Vim/neovim should have the feature though...

**Comment by blankname** [comment]
The major performance problem was caused by more highlights being defined each time `:Denite output:hi `was used.
Requiring two calls to `hi`, doesn't seem to have a large perf impact.
Performance seems acceptable now.

The current version seems to have introduced highlighting oddities to a few specific highlight entries:
image

**Comment by Shougo** [comment]
Hm...
The highlight seems broken.
I think the previous version is better.

https://files.gitter.im/Shougo/denite.nvim/ijNm/thumb/denite-hi.jpg

**Comment by Shougo** [comment]
The implementation is better.
But the highlight is worse...
Hm.  Pending now.

**Comment by blankname** [comment]
The latest commit should fix the highlighting regression.

Remaining issues:
The `Msg` in `ErrorMsg` not being highlighted correctly.
Many highlight attributes not being highlighted correctly.

**Comment by blankname** [comment]
I believe all major issues have been fixed:
image

**Comment by Shougo** [comment]
Merged.

**Pull request merged**